waiter/src/waiter/core.clj

```diff
@@ -1208,6 +1211,13 @@
                                                   :resolved-factory-fn resolved-factory-fn})))
                                reporter-instance))
                            codahale-reporters))
+   :event-maintainer (pc/fnk
```

Comment:
I think `instance-tracker` is a better name, `event` is too generic. It will need additional configuration for a predicate to trigger the notification. By default, the trigger can be a simple logging of instance counts. The configured method can do any filtering if it wants to skip notification events.